flink-tests/src/test/java/org/apache/flink/api/connector/source/lib/DataGeneratorSourceITCase.java

```diff
@@ -0,0 +1,255 @@
+import org.apache.flink.streaming.api.datastream.DataStreamSource;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.test.junit5.MiniClusterExtension;
+import org.apache.flink.util.TestLogger;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.LongStream;
+
+import static java.util.stream.Collectors.summingInt;
+import static org.apache.flink.core.testutils.FlinkAssertions.anyCauseMatches;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+
+/** An integration test for {@code DataGeneratorSource}. */
+public class DataGeneratorSourceITCase extends TestLogger {
+
+    private static final int PARALLELISM = 4;
+
+    @RegisterExtension
+    private static final MiniClusterExtension miniClusterExtension =
+            new MiniClusterExtension(
+                    new MiniClusterResourceConfiguration.Builder()
+                            .setNumberTaskManagers(1)
+                            .setNumberSlotsPerTaskManager(PARALLELISM)
+                            .build());
+
+    // ------------------------------------------------------------------------
+
+    @Test
+    @DisplayName("Combined results of parallel source readers produce the expected sequence.")
+    public void testParallelSourceExecution() throws Exception {
+        final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        env.setParallelism(PARALLELISM);
+
+        final DataStream<Long> stream = getGeneratorSourceStream(index -> index, env, 1_000L);
+
+        final List<Long> result = stream.executeAndCollect(10000);
+
+        assertThat(result).containsExactlyInAnyOrderElementsOf(range(0, 999));
+    }
+
+    @Test
+    @DisplayName("Generator function can be instantiated as an anonymous class.")
+    public void testParallelSourceExecutionWithAnonymousClass() throws Exception {
+        final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        env.setParallelism(PARALLELISM);
+
+        GeneratorFunction<Long, Long> generatorFunction =
+                new GeneratorFunction<Long, Long>() {
+
+                    @Override
+                    public Long map(Long value) {
+                        return value;
+                    }
+                };
+
+        final DataStream<Long> stream = getGeneratorSourceStream(generatorFunction, env, 1_000L);
+
+        final List<Long> result = stream.executeAndCollect(10000);
+
+        assertThat(result).containsExactlyInAnyOrderElementsOf(range(0, 999));
+    }
+
+    @Test
+    @DisplayName("Exceptions from the generator function are not 'swallowed'.")
```

Comment:
Hmm... In terms of navigation, I do not quite see what is the issue since double-clicking the failed test in IntelliJ immediately brings you to it, no manual matching is needed.
![image](https://user-images.githubusercontent.com/1492164/192537655-6b6d7033-48f2-41dd-88a1-ae2df8404a91.png)

And Maven prints the test name, just as before:
```
[ERROR] org.apache.flink.api.connector.source.lib.DataGeneratorSourceITCase.testFailingGeneratorFunctionInitialization  Time elapsed: 0.701 s  <<< FAILURE!
```
I find it more expressive, and I also see that we started adding it to the project in a couple of other places already, but we could also stick to the camelCaseTraditions if you think the `@DisplayName` is somehow detrimental.